### ISSUE TITLE
Null type in JSON schema ignore when generating swagger.

### ISSUE THREAD

**Issue by dormesica** [fastify/fastify-swagger#594]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.29.0

### Plugin version

6.0.1

### Node.js version

16.13.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.3.1

### Description

I'm using [Typebox](https://www.npmjs.com/package/@sinclair/typebox) to generate JSON schemas for an endpoint's response. In one endpoint some response properties may be `null`.

In Typebox we defined it as follow
```javascript
Type.Union([Type.String(), Type.Null()]
```
Which seems to generate the correct JSON schema
```json
{
  "anyOf": [
    { "type": "string" },
    { "type": "null" }
  ]
}
```

The generated swagger file does not indicate that field can be nullable (which cases some of our tests to fail).

### Steps to Reproduce

Create a schema with a property with type
```json
{
  "anyOf": [
    { "type": "string" },
    { "type": "null" }
  ]
}
```

Generate swagger file.

**Comment by climba03003** [comment]
It is expected. There is no type call `null` in `swagger` or `openapi`.
You should transform it on your own.

**Comment by dormesica** [comment]
Type can be declared as Nullable as far as I know.
It it not supported?

**Comment by climba03003** [comment]
> Type can be declared as Nullable as far as I know. It it not supported?

`"type": "null"` is the syntax of `JSON Schema`. Not `Swagger 2.0` or `OpenAPI 3`.
It is complicated for a library to determine how to transform it to proper one.

You should use `nullable` property for `Swagger 2.0` or `OpenAPI 3`. You may also need to add `keyword` for `ajv`.
```
Type.Union([Type.String({ nullable: true }), Type.Null()]

{
  "anyOf": [
    { "type": "string", nullable: true },
    { "type": "null" }
  ]
}
```

**Comment by dormesica** [comment]
I tried using the type  `Type.String({ nullable: true })` and `Type.Union([Type.String({ nullable: true }), Type.Null()]` but it didn't work.
The generated swagger still does not allow the property to be `null`.

**Comment by pebo** [comment]
> > Type can be declared as Nullable as far as I know. It it not supported?
> 
> `"type": "null"` is the syntax of `JSON Schema`. Not `Swagger 2.0` or `OpenAPI 3`. It is complicated for a library to determine how to transform it to proper one.
> 
> You should use `nullable` property for `Swagger 2.0` or `OpenAPI 3`. You may also need to add `keyword` for `ajv`.
> 
> ```
> Type.Union([Type.String({ nullable: true }), Type.Null()]
> 
> {
>   "anyOf": [
>     { "type": "string", nullable: true },
>     { "type": "null" }
>   ]
> }
> ```

Using the following versions:
```
    "@fastify/swagger": "^7.4.1",
    "@fastify/type-provider-typebox": "^2.3.0",
    "@sinclair/typebox": "^0.24.28",
    "fastify": "^4.5.2"
 ```
 
 Given a typebox definition like:
 ```
 deleted_at: Type.Union([Type.Null(), Type.String({ format: 'date-time',  nullable: true })]),
```
The following (invalid) open api 3.0 spec is generated:
```
deleted_at:
    anyOf:
      - type: 'null'
      - format: date-time
        nullable: true
        type: string
```
Dropping the Union would remove the in 3.0 invalid `type: 'null` part but that breaks the type validation as the attribute may be `null`. Is there a workaround for this issue?

**Comment by philsturgeon** [comment]
Just hopping in here to see how things are coming along, because if `type: null` is not supported that means OpenAPI v3.1 is fundamentally not supported, and with folks asking for OpenAPI v3.2 support last year (https://github.com/fastify/fastify-swagger/issues/906) I am trying to get a handle on whether this tool is still active. 

Fastify is an excellent framework, but either this issue is out of date or the tool has gone into what we call "Legacy", which is only support only OAS 3.0 or older.

Can anyone help me out? A good sign of showing it's a modern OpenAPI tool would be renaming it tbh, as Swagger is a long dead name which has not been relevant to OpenAPI other than SmartBear tools for almost a decade.

**Comment by RobvH** [comment]
@Tony133 True that in OpenAPI 3.0 you'd need `nullable: true`, but if `Type.Union([String, Null])` emits `{"anyOf":[…,{"type":"null"}]}`, then I think you're already fundamentally emitting 3.1 dialect. Handling JSONSchema's null was a Feb 2021 change that's part of the spec. Is there an intentional effort to stop supporting new specs? I find 3.2's tagging is epically useful and can tie into Fastify's schema tags pretty nicely.

Would you like a PR that makes these changes? This is a great tool, and to @philsturgeon's point, let's not allow this project to grow mold.

**Comment by Tony133** [comment]
> [@Tony133](https://github.com/Tony133) True that in OpenAPI 3.0 you'd need `nullable: true`, but if `Type.Union([String, Null])` emits `{"anyOf":[…,{"type":"null"}]}`, then I think you're already fundamentally emitting 3.1 dialect. Handling JSONSchema's null was a Feb 2021 change that's part of the spec. Is there an intentional effort to stop supporting new specs? I find 3.2's tagging is epically useful and can tie into Fastify's schema tags pretty nicely.
> 
> Would you like a PR that makes these changes? This is a great tool, and to [@philsturgeon](https://github.com/philsturgeon)'s point, let's not allow this project to grow mold.

@RobvH No, there are no plans to stop supporting the latest OpenAPI specifications — It's just that no one has dealt with it yet. If you have time and want to contribute, pull requests are always welcome 😉

If you notice me moving things around in the issues, it’s just that I’m currently tidying up issues across the Fastify ecosystem (duplicates, abandoned items, etc.) to figure out what’s actually worth working on. It’s unrelated to this issue. Feel free to open the PR whenever you like; I or another maintainer will review it as soon as possible.

**Comment by Tony133** [comment]
Fixed in #951 for OpenAPI 3.0 documents (3.1 already supported type: 'null' natively). Swagger 2.0 mode is unchanged: the 2.0 specification does not define nullable, anyOf/oneOf, the null type or the array form of type, so there is no valid form to convert these schemas to — a custom transform is still needed there.

Released in [v9.8.2](https://github.com/fastify/fastify-swagger/releases/tag/v9.8.2)
